File: NeuralNet/Regularization/L1Regularization.cs
using System;

namespace NeuralNet {
	public sealed class L1Regularization : Regularization {
		public L1Regularization(float regularizationFactor) : base(regularizationFactor) {}

		public override float GetDerivative(float value) {
			return RegularizationFactor*Math.Sign(value);
		}
	}
}

File: NeuralNetNative/L1Regularization.cpp
#define NEURALNETNATIVEAPI
#include <mathimf.h>
#include "L1Regularization.h"

namespace NeuralNetNative {
	L1Regularization::L1Regularization(float regularizationFactor) : Regularization(regularizationFactor) {
	}

	float L1Regularization::GetDerivative(float value) {
		return copysignf(RegularizationFactor, value);
	}
}